### ISSUE TITLE
Connecting payload not working when send authorization header

### ISSUE THREAD

**Issue by kietcthe** [apollographql/apollo-ios#3035]
### Question

Hi everyone,

This is the first time I use Apollo Client, so I don't know if this is a bug or something wrong with my code, but since I can't find the answer anywhere, I decided to create this question.

I currently use WebSocketTransport to create an ApolloClient, but when I pass connectingPayload to the WebSocketTransport init constructor, it will return me an error saying that I'm missing the authorization header.

```
var request = URLRequest(url: url)
let socketConnection = WebSocket(request: request, protocol: .graphql_ws)
let authPayload = ["Authorization": "Bearer \(authToken!)"]
let webSocketTransport = WebSocketTransport(websocket: socketConnection, config: WebSocketTransport.Configuration(connectingPayload: authPayload))
    
self.webSocketClient = ApolloClient(networkTransport: webSocketTransport, store: store)
```

**The error message:**
> "websocket is disconnected: WSError(type: ApolloWebSocket.WebSocket.WSError.ErrorType.protocolError, message: \"{\\\"server_error_msg\\\":\\\"4400: Connection initialization failed: Missing \\\'Authorization\\\' or \\\'Cookie\\\' header in JWT authenticati\", code: 4400)"
Error: The operation couldn’t be completed. (ApolloWebSocket.WebSocket.WSError error 1.)

But it will work when creating a URLRequest, adding headers and passing it to WebSocket, and then using that WebSocket instance to create WebSocketTransport.

```
var request = URLRequest(url: url)
request.setValue("Bearer \(authToken!)", forHTTPHeaderField: "Authorization")
let socketConnection = WebSocket(request: request, protocol: .graphql_ws)
let webSocketTransport = WebSocketTransport(websocket: socketConnection, config: WebSocketTransport.Configuration(connectingPayload: nil))
    
self.webSocketClient = ApolloClient(networkTransport: webSocketTransport, store: store)
```
Did I miss something or this is a bug with Apollo Client?

**Comment by kietcthe** [comment]
@BobaFetters 
I searched many issues and saw other people add Authorization and their token in the payload. That makes me curious. Maybe it depends on how the Websocket is implemented.

Thank you very much.

**Comment by BobaFetters** [comment]
@kietcthe yes, the implementation of the Websocket on the backend could affect this as well, in your case because of the error you are getting it is expecting the Authorization as a Header vs in the connecting payload.
